src/maintainer/knowledge_base.rst

```diff
@@ -1136,3 +1136,126 @@ key with ``docker_image``. Also ``cdt_name`` ensures the CDTs match the CentOS
+
+  Conda exposes the maximum CUDA version supported by the installed Nvidia drivers through a virtual package
+  named ``__cuda``. By default, ``conda`` will install the highest version available
+  for the packages involved. To override this behaviour, you can define a ``CONDA_OVERRIDE_CUDA`` environment
+  variable. More details in the
+  `Conda docs <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html#overriding-detected-packages>`_.
+
+  Note that prior to v4.8.4, ``__cuda`` versions would not be part of the constraints, so you would always
+  get the latest one, regardless the supported CUDA version.
+
+  If for some reason you want to install a specific version, you can use::
+
+    conda install your-gpu-package cudatoolkit=10.1
+
+Testing the packages
+--------------------
+
+Since the CI machines do not feature a GPU, you won't be able to test the built packages as part
+of the conda recipe. That does not mean you can't test your package locally. To do so:
+
+1. Enable the Azure artifacts for your feedstock (see :ref:`here <azure-config>`).
+2. Include the test files and requirements in the recipe
+   `like this <https://github.com/conda-forge/cupy-feedstock/blob/a1e9cdf47775f90d3153a26913068c6df942d54b/recipe/meta.yaml#L51-L61>`_.
+3. Provide the test instructions. Take into account that the GPU tests will fail in the CI run,
+   so you need to ignore them to get the package built and uploaded as an artifact.
+   `Example <https://github.com/conda-forge/cupy-feedstock/blob/a1e9cdf47775f90d3153a26913068c6df942d54b/recipe/run_test.py>`_.
+4. Once you have downloaded the artifacts, you will be able to run::
+
+    conda build --test <pkg file>.tar.bz2
+
+
+Common problems and known issues
+--------------------------------
+
+Helping your build system locate CUDA
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some build systems might need some help finding CUDA, so you might need to pass the value
+of ``CUDA_HOME`` as some kind of flag. Some examples include:
+
+* Old-style CMake (using the deprecated ``FindCUDA`` module), you might need to pass this flag::
+
+    -DCUDA_TOOLKIT_ROOT_DIR=${CUDA_HOME}
+
+* On Windows, if you need to pass ``CUDA_PATH``, CMake might choke on a backslash. In that case,
+  you can overwrite ``CUDA_PATH`` as follows::
+
+    set "CUDA_PATH=%CUDA_PATH:\=/%"
+
+* On Windows packages that are built with MSBuild, new-style CMake (using ``FindCUDAToolkit`` module)
+  you might need to define this environment variable::
+
+    set "CudaToolkitDir=%CUDA_PATH%"
+
+
+CUDA 11 and CDT packages
+^^^^^^^^^^^^^^^^^^^^^^^^
+If your package requires both CUDA and some CDTs on Linux, you will need to patch the
+``.ci_support/*.yml`` files corresponding to CUDA 11 and above so they reflect
+``cdt_name: cos7``. Do note that these changes do not survive feedstock rerenders!
+
+This should be fixed at some point, but for now you need to do it manually. Apologies!
```

Comment:
Can you send a PR to add this to the cuda11 migrator instead of the main pinning file? (That should workaround the earlier error)

Reply:
See https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1078